src/App.tsx

```diff
@@ -31,65 +31,125 @@ export function getReorderedGoods(
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods = visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+  }
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  reverseHandler = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  alphabeticSorter = () => {
+    this.setState({ sortType: SortType.ALPABET });
+  };
+
+  lengthSorter = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  resetHandler = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const {
+      isReversed,
+      sortType,
+    } = this.state;
+
+    const newGoods = getReorderedGoods(goodsFromServer,
+      { sortType, isReversed });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info ${sortType !== SortType.ALPABET && 'is-light'}`}
+            onClick={this.alphabeticSorter}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={`button is-success ${sortType !== SortType.LENGTH && 'is-light'}`}
+            onClick={this.lengthSorter}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={`button is-warning ${!isReversed && 'is-light'}`}
+            onClick={this.reverseHandler}
+          >
+            Reverse
+          </button>
+
+          {(sortType || isReversed)
+            ? (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={this.resetHandler}
+              >
+                Reset
+              </button>
+            )
+            : ''}
```

Comment:
Use && if you want to show or not show something
```suggestion
          {(sortType || isReversed)
            && (
              <button
                type="button"
                className="button is-danger is-light"
                onClick={this.resetHandler}
              >
                Reset
              </button>
            )}
```